src/main/java/uk/gov/companieshouse/chs/gov/uk/notify/integration/api/templatelookup/TemplateLookup.java

```diff
@@ -0,0 +1,30 @@
+package uk.gov.companieshouse.chs.gov.uk.notify.integration.api.templatelookup;
+
+import org.springframework.stereotype.Component;
+
+/**
+ * Responsible for inferring the template asset location from its key.
+ */
+@Component
+public class TemplateLookup {
+
+    private static final String LETTER_TEMPLATES_ROOT_DIRECTORY = "assets/templates/letters/";
+
+    /**
+     * Derives a specification of the template location from the template look up key
+     * provided.
+     * @param templateLookupKey key or specification consisting of the client application ID
+     *                          (aka service name),
+     * @return the inferred location of the template
+     */
+    public LetterTemplateLocatorSpec lookupTemplate(ChLetterTemplate templateLookupKey) {
```

Comment:
Nit pick but `templateLookupKey` makes it look like this is the key itself - which I think it is? But might be good to just name it `ChLetterTemplateKey` so it doesn't at first glance look like an object holding the letter template itself?

```suggestion
    public LetterTemplateLocatorSpec lookupTemplate(ChLetterTemplateKey templateLookupKey) {
```